sdk/invoke.js

```diff
@@ -21,10 +21,25 @@ const invoke = async function invoke(funcName, apiKey, funcData) {
     resolveWithFullResponse: true,
   };
   logger.debug('Invoking function', options);
-  const res = await rp.post(options);
-  const { statusCode, headers, body } = res;
-  logger.debug('Invoke response', { statusCode, headers, body });
-  return res;
+  try {
+    const res = await rp.post(options);
+    const { statusCode, headers, body } = res;
+    logger.debug('Invoke response', { statusCode, headers, body });
+    return res;
+  } catch (err) {
+    let parsedError;
+    try {
+      parsedError = JSON.parse(err.error);
+    } catch (nestedErr) {
+      throw new Error(err.error || err);
```

Comment:
Should probably be combined with the default throw after the ifs, and should consider adding text here (or in the internal system errors) to differentiate the cases from user perspective.
Is this covered by the spec test below?

Reply:
Definitely could be merged. It is tested by the cases below such as with Java8.